pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -22,77 +22,101 @@ import (
 
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/kubelet/cm/dra/state"
 	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
 )
 
 // claimInfo holds information required
 // to prepare and unprepare a resource claim.
 type claimInfo struct {
-	sync.RWMutex
-
-	// name of the DRA driver
-	driverName string
-
-	// claimUID is an UID of the resource claim
-	claimUID types.UID
-
-	// claimName is a name of the resource claim
-	claimName string
-
-	// namespace is a claim namespace
-	namespace string
-
-	// podUIDs is a set of pod UIDs that reference a resource
-	podUIDs sets.Set[string]
-
-	// cdiDevices is a list of CDI devices returned by the
-	// GRPC API call NodePrepareResource
-	cdiDevices []string
-
+	state.ClaimInfoState
 	// annotations is a list of container annotations associated with
 	// a prepared resource
 	annotations []kubecontainer.Annotation
 }
 
-func (res *claimInfo) addPodReference(podUID types.UID) {
-	res.Lock()
-	defer res.Unlock()
-
-	res.podUIDs.Insert(string(podUID))
-}
-
-func (res *claimInfo) deletePodReference(podUID types.UID) {
-	res.Lock()
-	defer res.Unlock()
-
-	res.podUIDs.Delete(string(podUID))
-}
-
 // claimInfoCache is a cache of processed resource claims keyed by namespace + claim name.
 type claimInfoCache struct {
 	sync.RWMutex
+	state     state.CheckpointState
 	claimInfo map[string]*claimInfo
 }
 
+func newClaimInfo(driverName string, claimUID types.UID, claimName, namespace, podUID string, cdiDevice []string) (*claimInfo, error) {
+
+	// NOTE: Passing CDI device names as annotations is a temporary solution
+	// It will be removed after all runtimes are updated
+	// to get CDI device names from the ContainerConfig.CDIDevices field
+	annotations, err := generateCDIAnnotations(claimUID, driverName, cdiDevice)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate container annotations, err: %+v", err)
+	}
+	claimInfoState := state.ClaimInfoState{
+		DriverName: driverName,
+		ClaimUID:   claimUID,
+		ClaimName:  claimName,
+		Namespace:  namespace,
+		PodUIDs:    sets.New(podUID),
+		CdiDevices: cdiDevice,
+	}
+	claimInfo := claimInfo{ClaimInfoState: claimInfoState, annotations: annotations}
+	return &claimInfo, nil
+}
+
 // newClaimInfoCache is a function that returns an instance of the claimInfoCache.
-func newClaimInfoCache() *claimInfoCache {
-	return &claimInfoCache{
-		claimInfo: make(map[string]*claimInfo),
+func newClaimInfoCache(stateDir, checkpointName string) (*claimInfoCache, error) {
+
+	stateImpl, err := state.NewCheckpointState(stateDir, checkpointName)
+	if err != nil {
+		return nil, fmt.Errorf("Could not initialize checkpoint manager, please drain node and remove dra state fil, err: %+v", err)
+	}
+
+	curState, err := stateImpl.GetOrCreate()
+	if err != nil {
+		return nil, err
```

Comment:
please wrap this error with something like: "error calling GetOrCreate() on checkpoint state: %v".

Reply:
You've resolved this, but the error is still not wrapped...